polars-pipe 0.30.0 won't compile

### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.


### Issue description

Originally I was running into this issue: https://github.com/pola-rs/polars/issues/9125
I followed the thread, which told me to delete my lockfile and recompile. Now I'm hit with the following error:
```rust
    Checking polars-pipe v0.30.0
error[E0308]: `match` arms have incompatible types
   --> C:\Users\me\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-pipe-0.30.0\src\pipeline\convert.rs:155:13
    |
127 |       let out = match lp_arena.get(node) {
    |                 ------------------------ `match` arms have incompatible types
...
131 | /             match &payload.file_type {
132 | |                 #[cfg(feature = "parquet")]
133 | |                 FileType::Parquet(options) => {
134 | |                     Box::new(ParquetSink::new(path, *options, input_schema.as_ref())?)
...   |
141 | |                 FileType::Memory => Box::new(OrderedSink::new()),
142 | |             }
    | |_____________- this is found to be of type `Box<ordered::OrderedSink>`
...
155 | /             match &options.how {
156 | |                 #[cfg(feature = "cross_join")]
157 | |                 JoinType::Cross => {
158 | |                     Box::new(CrossJoin::new(options.suffix.clone())) as Box<dyn Sink>
...   |
192 | |                 _ => unimplemented!(),
193 | |             }
    | |_____________^ expected `Box<OrderedSink>`, found `Box<GenericBuild>`
    |
    = note: expected struct `Box<ordered::OrderedSink>`
               found struct `Box<generic_build::GenericBuild>`

```

### Reproducible example

Not sure how to give a repro; I'm just compiling polars 0.30.0

### Expected behavior

Expected to compile succesfully

### Installed versions

<details>
    "abs",
    "avx512",
    "cum_agg",
    "dtype-slim",
    "fmt",
    "lazy",
    "object",
    "performant",
    "random",
    "rolling_window",
    "temporal",
</details>

--- a/polars/polars-lazy/polars-pipe/src/pipeline/convert.rs
+++ b/polars/polars-lazy/polars-pipe/src/pipeline/convert.rs
@@ -138,7 +138,7 @@ where
                 FileType::Ipc(options) => {
                     Box::new(IpcSink::new(path, *options, input_schema.as_ref())?) as Box<dyn Sink>
                 }
-                FileType::Memory => Box::new(OrderedSink::new()),
+                FileType::Memory => Box::new(OrderedSink::new()) as Box<dyn Sink>,
             }
         }
         Join {

